src/status_im2/contexts/communities/menus/request_to_join/view.cljs

```diff
@@ -123,8 +123,8 @@
          {:accessibility-label :join-community-button
           :on-press            (fn []
                                  (if can-join?
-                                   (do (rf/dispatch [::communities/join id])
-                                       (rf/dispatch [:bottom-sheet/hide]))
+                                   (rf/dispatch [:bottom-sheet/hide-and-dispatch
```

Comment:
im not sure i understand the point of this change, i believe we want to dispatch event as soon as possible and do not wait for bottom sheet closing

Reply:
yeah then we need to fix bottom sheet, because it's just a hack, we shouldn't care about that in the code, bottom sheet should care about that itself

Reply:
yes i mean we can't make a rule and make all other code more complex, there will be different cases when new bottom sheet will be opened async without closing the current one, so bottom sheet should take care of this case

Reply:
it should be simple to implement